productivity-suite/app/fragments/todo-lists/src/root.tsx

```diff
@@ -44,17 +46,36 @@ export const Root = component$(() => {
 
   const ref = useSignal<Element>();
 
+  useClientEffect$(() => {
+    if (ref.value) {
+      return getBus(ref.value).listen({
+        eventName: "todos-updated",
+        callback: ({
+          listName,
+          todos,
+        }: {
+          listName: string;
+          todos: Todo[];
+        }) => {
+          const list = state.todoLists.find(({ name }) => name === listName);
+          if (list) {
+            list.todos = todos;
+          }
+        },
+      });
+    }
```

Comment:
Why is this listener here?

Reply:
ah yeah, good spot! 😁 

so basically we want the `todo-lists` to be up to date with the todos, because when you select a list, it dispatches an event with the whole list and its todos, which is then taken by the todos fragment which uses that data. If the `todo-lists` wouldn't keep the todos up to date then the todos fragment would receive old outdated todos when lists are selected.

I found this to be a sort of ok, I also thought of other solutions (which seemed slightly worse to me):
- having the `todo-list-selected` only include the list name and then whoever needs the todos would manually "fetch them from the backend" (/take them from the cookies)
- having the todos fragment keep the whole list of lists in its state (currently it only keeps the current one since it has no need for the rest), so that it can ignore the todos coming from the even and use those that is has in its state (the event can also only include the list name here as well)

what do you think? does one of the other solution look better to you? or do you have a different idea/suggestion? 🙂

Reply:
It feels like the `todos-list` fragment should only care about the todo list names.
And the `todos` fragment should only care about the todo items in the list it is displaying.

Reply:
Yeah, I think so. Micro-frontends should be pretty self-contained and not really rely on other fragments (or even their container) to provide them with data.

Reply:
There could be some cleverness done with Service Workers / Durable Objects / Caching / GraphQL that might avoid duplicated fetches...

Reply:
I've moved changed the events so that only the list name is being dispatched and the todos are fetched if needed 🙂, so it should be all good 🙂